src/dateparser/dateparser.js

```diff
@@ -82,6 +108,7 @@ angular.module('ui.bootstrap.dateparser', [])
     }
 
     format = $locale.DATETIME_FORMATS[format] || format;
+    format = format.replace(SPECIAL_CHARACTERS_REGEXP, '\\$&');
```

Comment:
We should instead should do this right before the `new RegExp()` call inside `createParser` as that shows why this character replacement is needed. There's also less weird characters to process in the `createParser`.